src/main/java/wolox/training/controllers/BookController.java

```diff
@@ -0,0 +1,73 @@
+package wolox.training.controllers;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.http.HttpStatus;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+import org.springframework.web.server.ResponseStatusException;
+import wolox.training.exceptions.BookNotFoundException;
+import wolox.training.exceptions.BookIdMismatchException;
+import wolox.training.repositories.BookRepository;
+import wolox.training.models.Book;
+
+import java.util.List;
+
+@RestController
+@RequestMapping("/api/books")
+public class BookController {
+
+    @Autowired
+    private BookRepository bookRepository;
+
+    @GetMapping
+    public Iterable findAll() {
+        return bookRepository.findAll();
+    }
+
+    @GetMapping("/title/{bookTitle}")
+    public List findByTitle(@PathVariable String bookTitle) {
+        try {
+            return bookRepository.findByTitle(bookTitle);
+        } catch (BookNotFoundException ex) {
+            throw new ResponseStatusException(
+                    HttpStatus.NOT_FOUND, "Book Not Found", ex);
+        }
+    }
+
+    @GetMapping("/{id}")
+    public Book findOne(@PathVariable Long id) {
```

Comment:
Use final here, and in all variables that won't change. Check the benefits of using final https://softwareengineering.stackexchange.com/questions/98691/excessive-use-final-keyword-in-java